packages/excalidraw/components/App.tsx

```diff
@@ -6158,11 +6161,32 @@ class App extends React.Component<AppProps, AppState> {
       }
     }
 
-    const hitElement = this.getElementAtPosition(
-      scenePointer.x,
-      scenePointer.y,
+    const hitElementMightBeLocked = this.getElementAtPosition(
+      scenePointerX,
+      scenePointerY,
+      {
+        includeLockedElements: true,
+      },
     );
 
+    const hitElements = this.getElementsAtPosition(
+      scenePointerX,
+      scenePointerY,
+    );
+
+    let hitElement: ExcalidrawElement | null = null;
+    if (
+      hitElementMightBeLocked &&
+      hitElementMightBeLocked.locked &&
+      !hitElements.some((el) => this.state.selectedElementIds[el.id])
+    ) {
+      hitElement = null;
+    } else if (hitElementMightBeLocked && !hitElementMightBeLocked.locked) {
+      hitElement = hitElementMightBeLocked;
+    } else {
+      hitElement = this.getElementAtPosition(scenePointerX, scenePointerY);
```

Comment:
the last branch here may just filter out the locked elements instead, which should be faster than re-doing the hit checks?

either way, I'm not see keen on this whole block as we now do the expernsive `getElementsAtPosition()` effectively 2x even after the 3rd branch hit check removal.